hir/src/component/mod.rs

```diff
@@ -0,0 +1,284 @@
+use core::{
+    convert::{AsMut, AsRef},
+    ops::{Deref, DerefMut},
+};
+use intrusive_collections::RBTree;
+use std::collections::BTreeMap;
+
+use super::*;
+
+mod interface;
+
+pub use interface::*;
+
+/// Represents the method by which a component function should be invoked in Miden VM
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum FunctionInvocationMethod {
+    /// A function should be invoked by a `call` Miden VM instruction
+    Call,
+    /// A function should be invoked by a `exec` Miden VM instruction
+    #[default]
+    Exec,
+}
+
+/// A component import
+#[derive(Debug)]
+pub struct ComponentImport {
+    /// The interfact function name that is being imported
+    pub interface_function: InterfaceFunctionIdent,
+    /// The component(lifted) type of the imported function
+    pub function_ty: LiftedFunctionType,
+    /// The method of calling the function
+    pub invoke_method: FunctionInvocationMethod,
+    /// The MAST root hash of the function to be used in codegen
+    pub function_mast_root_hash: MastRootHash,
+}
+
+/// The name of a exported function
+#[derive(Debug, Ord, PartialEq, PartialOrd, Eq, Hash, derive_more::From, derive_more::Into)]
+pub struct FunctionExportName(Symbol);
+
+/// A component export
+#[derive(Debug)]
+pub struct ComponentExport {
```

Comment:
Should this type have an optional `digest` field as well? Obviously exports which reference functions that haven't been compiled yet won't have one, but if there are re-exported functions, those might actually have MAST roots associated with them. But perhaps this type only ever refers to functions that aren't compiled yet, in which case, feel free to disregard.

Reply:
I don't think so. It's built from the core module export, which, I guess, does not support re-exports (i.e., export an imported function).